pkg/crc/preflight/preflight_check_tray_darwin.go

```diff
@@ -0,0 +1,248 @@
+		return errors.New("crc daemon is not running")
+	}
+	logging.Debug("Checking if tray plist file exists")
+	if !plistFileExists(trayPlistFilePath) {
+		return errors.New("Tray plist file does not exist")
+	}
+	logging.Debug("Checking if tray agent running")
+	if !agentRunning(trayAgentLabel) {
+		return errors.New("Tray is not running")
+	}
+	return nil
+}
+
+func fixTrayExistsAndRunning() error {
+	// get the tray app
+	if !trayAppCached() {
+		err := downloadOrExtractTrayApp()
+		if err != nil {
+			return err
+		}
+	}
+	currentExecutablePath, err := goos.Executable()
+	if err != nil {
+		return err
+	}
+	daemonConfig := AgentConfig{
+		BinaryPath:     currentExecutablePath,
+		StdOutFilePath: stdOutFilePathDaemon,
+	}
+
+	trayConfig := AgentConfig{
+		BinaryPath:     filepath.Join(constants.CrcBinDir, constants.TrayBinaryName, "Contents", "MacOS", "CodeReady Containers"),
+		StdOutFilePath: stdOutFilePathTray,
+	}
+	logging.Debug("Creating daemon plist")
+	err = createPlist(daemonPlistTemplate, daemonConfig, daemonPlistFilePath)
+	if err != nil {
+		return err
+	}
+	logging.Debug("Creating tray plist")
+	err = createPlist(trayPlistTemplate, trayConfig, trayPlistFilePath)
+	if err != nil {
+		return err
+	}
+
+	// load crc daemon
+	err = loadPlist(daemonPlistFilePath)
+	if err != nil {
+		return err
+	}
+	if !agentRunning(daemonAgentLabel) {
+		if err = startAgent(daemonAgentLabel); err != nil {
+			return err
+		}
+	}
+	// load tray
+	err = loadPlist(trayPlistFilePath)
+	if err != nil {
+		return err
+	}
+	if !agentRunning(trayAgentLabel) {
+		if err = startAgent(trayAgentLabel); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func createPlist(template string, config AgentConfig, plistPath string) error {
+	var plistContent bytes.Buffer
+	t, err := goTemplate.New("plist").Parse(template)
+	if err != nil {
+		return err
+	}
+	err = t.Execute(&plistContent, config)
+	if err != nil {
+		return err
+	}
+	err = ioutil.WriteFile(plistPath, plistContent.Bytes(), 0644)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func loadPlist(plistFilePath string) error {
+	_, err := exec.Command("launchctl", "load", plistFilePath).Output() // #nosec G204
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func startAgent(label string) error {
+	_, err := exec.Command("launchctl", "start", label).Output() // #nosec G204
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// check if a service (daemon,tray) is running
+func agentRunning(label string) bool {
+	// This command return a PID if the process
+	// is running, otherwise returns "-"
+	launchctlListCommand := `launchctl list | grep %s | awk '{print $1}'`
+	cmd := fmt.Sprintf(launchctlListCommand, label)
+	out, err := exec.Command("bash", "-c", cmd).Output() // #nosec G204
+	if err != nil {
+		return false
+	}
+	if strings.TrimSpace(string(out)) == "-" {
+		return false
+	}
+	return true
+}
+
+func plistFileExists(plistFilePath string) bool {
+	return os.FileExists(plistFilePath)
```

Comment:
why the indirection/helper? We could directly use `os.FileExists(plistFilePath)`  throughout this file?

Reply:
Right, let me fix this!